src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -803,262 +826,267 @@ void func_8089BD28(EnDinofos* this, PlayState* play) {
+            EnDinofos_IsFacingPlayer(this) && (Rand_ZeroOne() < 0.6f)) {
+            EnDinofos_SetupStartBreatheFire(this);
         } else {
-            func_8089A9B0(this, play);
+            EnDinofos_ChooseAction(this, play);
         }
     }
 }
 
-void func_8089C4F8(EnDinofos* this) {
+void EnDinofos_SetupSlash(EnDinofos* this) {
     Animation_PlayOnce(&this->skelAnime, &gDinolfosAttackAnim);
-    this->colliderQuad.base.atFlags &= ~AT_BOUNCED;
+    this->knifeCollider.base.atFlags &= ~AT_BOUNCED;
     Actor_PlaySfx(&this->actor, NA_SE_EN_RIZA_CRY);
-    this->unk_290 = 0;
-    this->unk_292 = -1;
+    this->attackTimer = 0;
+    this->timer2 = -1;
     this->actor.speed = 0.0f;
     this->actor.world.rot.y = this->actor.shape.rot.y;
-    this->actionFunc = func_8089C56C;
+    this->actionFunc = EnDinofos_Slash;
 }
 
-void func_8089C56C(EnDinofos* this, PlayState* play) {
-    this->unk_290++;
+void EnDinofos_Slash(EnDinofos* this, PlayState* play) {
+    this->attackTimer++;
     Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x800);
     this->actor.world.rot.y = this->actor.shape.rot.y;
     if (SkelAnime_Update(&this->skelAnime)) {
-        EffectBlure_AddSpace(Effect_GetByIndex(this->unk_2A0));
-        this->colliderQuad.base.atFlags &= ~AT_ON;
-        func_8089A9B0(this, play);
+        EffectBlure_AddSpace(Effect_GetByIndex(this->effectIndex));
+        this->knifeCollider.base.atFlags &= ~AT_ON;
+        EnDinofos_ChooseAction(this, play);
     } else if (Animation_OnFrame(&this->skelAnime, 7.0f)) {
-        this->colliderQuad.base.atFlags |= AT_ON;
-        func_8089AD70(this);
+        this->knifeCollider.base.atFlags |= AT_ON;
+        EnDinofos_EnableBumperCollision(this);
     } else if (Animation_OnFrame(&this->skelAnime, 13.0f)) {
-        this->colliderQuad.base.atFlags &= ~AT_ON;
+        this->knifeCollider.base.atFlags &= ~AT_ON;
     } else if (Animation_OnFrame(&this->skelAnime, 11.0f)) {
         Actor_PlaySfx(&this->actor, NA_SE_EN_RIZA_ATTACK);
     } else if (this->skelAnime.curFrame < 7.0f) {
-        func_8089AE00(this, play);
+        EnDinofos_Dodge(this, play);
     }
 }
 
-void func_8089C690(EnDinofos* this) {
+void EnDinofos_SetupRecoil(EnDinofos* this) {
     Animation_Change(&this->skelAnime, &gDinolfosAttackAnim, -1.0f, this->skelAnime.curFrame, 0.0f, ANIMMODE_ONCE,
                      0.0f);
-    this->colliderQuad.base.atFlags &= ~(AT_ON | AT_BOUNCED);
-    if (this->actionFunc != func_8089C2A8) {
+    this->knifeCollider.base.atFlags &= ~(AT_ON | AT_BOUNCED);
+    if (this->actionFunc != EnDinofos_JumpSlash) {
         this->actor.world.rot.y = this->actor.shape.rot.y;
     } else {
         this->actor.speed = 3.0f;
     }
-    this->actionFunc = func_8089C724;
+    this->actionFunc = EnDinofos_RecoilFromBlockedSlash;
 }
 
-void func_8089C724(EnDinofos* this, PlayState* play) {
+void EnDinofos_RecoilFromBlockedSlash(EnDinofos* this, PlayState* play) {
```

Comment:
```suggestion
void EnDinofos_Recoil(EnDinofos* this, PlayState* play) {
```
This should also be updated :) .

Reply:
What? This one I'm sure I did. Very odd. Will be in next commit.